### PR TITLE
chore: frontend refactor

### PR DESCRIPTION
**Pull request by DavideIadeluca**
**Fixes #0000**

**Changes proposed in this pull request:**

**Reviewers should focus on:**

**Screenshot**

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).

**Required changes:**

- [ ] Related [Flarum core extension PR's](https://github.com/flarum/core/pulls): (Omit this section if irrelevant)

### CODE REVIEW

File: js/src/forum/extend.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Extend from 'flarum/common/extenders';
+import Discussion from 'flarum/common/models/Discussion';
+
+export default [
+  new Extend.Model(Discussion) //
+    .attribute<boolean>('frontpage')
+    .attribute<string>('frontdate')
```
Comment (imorland) [inline js/src/forum/extend.ts]: Should this not be `Date|null`? 🤔

Comment (DavideIadeluca) [reply js/src/forum/extend.ts]: I was thinking so as well, but the backend returns a String

Comment (imorland) [reply js/src/forum/extend.ts]: Perhaps we should cast this on the backend model in that case, or might that be a breaking change I wonder. Will check shortly

Comment (DavideIadeluca) [reply js/src/forum/extend.ts]: Actually, I'm not so sure anymore where exactly I saw that it's returning a string. I can take a look at it again if you want

Comment (DavideIadeluca) [reply js/src/forum/extend.ts]: @imorland I had another look, we are casting it as `datetime` in the backend, but PHP returns a string in the ISO 8601 format so it's a `string` in the frontend. We could convert it to a JS Date, but I'm not sure if that's something we want to do?

Comment (imorland) [reply js/src/forum/extend.ts]: That's perfect, in this case, I would take the example from here, and define it like this:

https://github.com/FriendsOfFlarum/best-answer/blob/b1c621093e284f7385f15bc0c1ce1a00f1a1291a/js/src/forum/extend.ts#L17

Comment (DavideIadeluca) [reply js/src/forum/extend.ts]: Yep, makes sense.. changed in https://github.com/FriendsOfFlarum/frontpage/pull/32/commits/f65e9743d7b02703ac5bc776953fdd5bf13ef7f1

**Review by imorland — APPROVED** [review]

**Pull request merged**